lib/lte_link_control/lte_lc_helpers.c

```diff
@@ -435,6 +435,141 @@ int parse_edrx(const char *at_response, struct lte_lc_edrx_cfg *cfg)
 	return err;
 }
 
+/* Differenet values in the T3324 lookup table. */
+#define T3324_LOOKUP_DIFFERENT_VALUES 3
+/* Differenet values in the T3412 extended lookup table. */
+#define T3412_EXT_LOOKUP_DIFFERENT_VALUES 7
+/* Maximum value for the timer value field of the timer information elements in both
+ * T3324 and T3412 extended lookup tables.
+ */
+#define TIMER_VALUE_MAX 31
+
+/* Lookup table for T3324 timer used for PSM active time in seconds.
+ * Ref: GPRS Timer 2 IE in 3GPP TS 24.008 Table 10.5.163/3GPP TS 24.008.
+ */
+static const uint32_t t3324_lookup[8] = {2, 60, 360, 60, 60, 60, 60, 0};
+
+/* Lookup table for T3412-extended timer used for periodic TAU. Unit is seconds.
+ * Ref: GPRS Timer 3 in 3GPP TS 24.008 Table 10.5.163a/3GPP TS 24.008.
+ */
+static const uint32_t t3412_ext_lookup[8] = {600, 3600, 36000, 2, 30, 60, 1152000, 0};
+
+/* Lookup table for T3412 (legacy) timer used for periodic TAU. Unit is seconds.
+ * Ref: GPRS Timer in 3GPP TS 24.008 Table 10.5.172/3GPP TS 24.008.
+ */
+static const uint32_t t3412_lookup[8] = {2, 60, 360, 60, 60, 60, 60, 0};
```

Comment:
Could the tables be defined without fixed size:

`static const uint32_t t3324_lookup[] = {2, 60, 360};`

And the number of elements determined with the ARRAY_SIZE macro?

`encode_psm_timer(active_time_str, rat, t3324_lookup, ARRAY_SIZE(t3324_lookup));`

Reply:
> Could the tables be defined without fixed size:
> 
> `static const uint32_t t3324_lookup[] = {2, 60, 360};`
> 
> And the number of elements determined with the ARRAY_SIZE macro?
> 
> `encode_psm_timer(active_time_str, rat, t3324_lookup, ARRAY_SIZE(t3324_lookup));`

I'm not using the array size as a parameter for `encode_psm_timer()`. I could but it made sense to me to just handle the values that are different in the tables.

Reply:
But why there are numbers which are not used in the tables? I'm not following. 😄

Reply:
They are all used when parsing (decoding) to comply with 3GPP spec but they are not needed when encoding.
So yes, ARRAY_SIZE is already used in parsing but I'd rather use defines that I've introduced in the encoding side.

The tables are not new. `[8]` could be `[]` but I didn't change these from what they have originally been by @jtguggedal .

Reply:
Ok, I didn't see that those were already present earlier, just moved to a different place. I guess the decoding shouldn't need the extra values either, but the implementation would obviously have to be changed also there.

Reply:
Decoding needs those extra values to easily support the spec:
"Other values shall be interpreted as multiples of 1 minute in this version of the protocol."

Reply:
Please resolve this if there are no further questions.

Reply:
I'm not sure if modem would actually ever send such values, even if received from the network. Modem decodes the value and interprets the length of the timer, I don't think it should send such illegal values in the AT interface. The comment you copied is from the 3GPP specification, not from the AT command spec. But anyway, this could be reworked in another PR.